### DISCUSSION TITLE
Run any shell commands using OS shell in Tauri App Env

### DISCUSSION THREAD

**Discussion by jahanzeb10** [tauri-apps/tauri#3600] [Q&A] [answered]
I've setup a basic tauri app for my project and testing some of its apis like shell, cli etc. My question is:
Is it possible to run any commands in OS command line shell like basic `ls -la`, on a simple action like button click in Tauri App dev environment?

I've read [shell JS api](https://tauri.studio/docs/api/js/modules/shell) but getting some erros like : 
- `tauri.conf.json` error on `tauri > allowlist > shell`: Additional property 'scope' is not allowed 

Here's tauri.conf.json: 
"allowlist": {
      "all": true,
      "shell": {
        "scope": [
          {
            "name": "ls",
            "cmd": "ls",
            "args": ["-la"]
          }
        ]
    },
   }
   
A basic react function:
```
  const runLS = async () => {
    new Command('ls', ['-la'])
  }
```

 HTML: `<div onClick={runLS}>test</div>`
 
I haven't found any relevant example or discussions and documentation is not that detailed. Help!

**Comment by Kumassy** [comment, marked as answer]
I ran into the same error.

Please make sure you are using the latest tauri version.
Shell scope functions seems relative new feature, which was implemented in https://github.com/tauri-apps/tauri/commit/10314cd5cfe956fe5b3ae10acf551a92bdc366d3

Updating 

- from `"@tauri-apps/api": "^1.0.0-beta.8"` to `"@tauri-apps/api": "^1.0.0-rc.2`
- from `"@tauri-apps/cli": "^1.0.0-beta.10"` to `"@tauri-apps/cli": "^1.0.0-rc.7"`
- from `tauri-build = { version = "1.0.0-beta.4" }` to `tauri-build = { version = "1.0.0-rc.4" }`
- from `tauri = { version = "1.0.0-beta.8", features = ["api-all"] }` to `tauri = { version = "1.0.0-rc.4", features = ["api-all"] }`

in `package.json` and `src-tauri/Cargo.toml` solved the issue in my case.

This example demonstrates how to use `Command` API in JavaScript and how to configure `tauri.conf.json`: https://github.com/tauri-apps/tauri/tree/dev/examples/api
